allennlp/data/dataset_readers/conll2003.py

```diff
@@ -95,12 +103,20 @@ def __init__(
         for label in feature_labels:
             if label not in self._VALID_LABELS:
                 raise ConfigurationError("unknown feature label type: {}".format(label))
-        if coding_scheme not in ("IOB1", "BIOUL"):
-            raise ConfigurationError("unknown coding_scheme: {}".format(coding_scheme))
+        if convert_to_coding_scheme not in (None, "BIOUL") or coding_scheme not in (
+            "IOB1",
+            "BIOUL",
+        ):
+            raise ConfigurationError("unknown coding_scheme: {}".format(convert_to_coding_scheme))
+        if convert_to_coding_scheme is None:
```

Comment:
I'm not sure when I should show a warning to users. The current implementation would warn when `coding_scheme` is `IOB1` (default value) and neither `coding_scheme` nor `convert_to_coding_scheme` is specified. The default value of `convert_to_coding_scheme` is `None`, which means users using `Conll2003DatasetReader` with default coding scheme (`IOB1`) always see the warning (users may be frustrated...).

Reply:
Actually, I'd recommend a slightly different approach:

Entirely remove the "coding_scheme" parameter, but check `kwargs` for "coding_scheme". If `"coding_scheme" is given in `kwargs`, pop it out and then issue this warning.